Analysis/tasks.py

```diff
@@ -0,0 +1,152 @@
+        hist_config = self.hist_config
+        unc_config = os.path.join(self.ana_path(), 'config', 'weight_definition.yaml')
+        sample_config = self.sample_config
+        HistProducerFile = os.path.join(self.ana_path(), 'Analysis', 'HistProducerFile.py')
+        outDir = os.path.join(self.central_Histograms_path(), sample_name, 'tmp')
+        HistProducerFile_cmd = ['python3', HistProducerFile,'--inFile', input_file, '--outFileName',outFileName, '--dataset', sample_name, '--outDir', outDir,
+                            '--compute_unc_variations', 'True', '--compute_rel_weights', 'True',
+                            '--uncConfig', unc_config, '--histConfig', hist_config,'--sampleConfig', sample_config]
+        sh_call(HistProducerFile_cmd,verbose=1)
+
+
+class HistProducerSampleTask(Task, HTCondorWorkflow, law.LocalWorkflow):
+    max_runtime = copy_param(HTCondorWorkflow.max_runtime, 1.0)
+
+    def workflow_requires(self):
+        return { "histProducerFile" : HistProducerFileTask.req(self, branch=-1, branches=())}
+
+    def requires(self):
+        histProducerFile_map = HistProducerFileTask.req(self, branch=-1, branches=()).create_branch_map()
+        sample_name = self.branch_data
+        reqs = []
+        for br_idx, (smpl_name, prod_br) in histProducerFile_map.items():
+            if smpl_name == sample_name:
+                reqs.append( HistProducerFileTask.req(self, branch=br_idx, max_runtime=HistProducerFileTask.max_runtime._default, branches=()))
+        return reqs
+
+    def create_branch_map(self):
+        n = 0
+        branches = {}
+        already_considered_samples = []
+        histProducerFile_map = HistProducerFileTask.req(self, branch=-1, branches=()).create_branch_map()
+        for br_idx, (sample_name, prod_br) in histProducerFile_map.items():
+            if sample_name in already_considered_samples:
+                continue
+            branches[n] = sample_name
+            already_considered_samples.append(sample_name)
+            n+=1
+        return branches
+
+    def output(self):
+        sample_name = self.branch_data
+        vars_to_plot = list(self.hists.keys())
+        local_files_target = []
+        for var in vars_to_plot:
+            fileName = f'{var}.root'
+            outDir = os.path.join(self.central_Histograms_path(), sample_name)
+            if not os.path.isdir(outDir):
+                os.makedirs(outDir)
+            outFile = os.path.join(outDir,fileName)
+            local_files_target.append(law.LocalFileTarget(outFile))
+        return local_files_target
+
+    def run(self):
+        sample_name = self.branch_data
+        histProducerFile_map = HistProducerFileTask.req(self, branch=-1, branches=()).create_branch_map()
+        files_idx = []
+        hist_config = os.path.join(self.ana_path(), 'config', 'plot','histograms.yaml')
+        vars_to_plot = list(self.hists.keys())
+        hists_str = ','.join(var for var in vars_to_plot)
```

Comment:
- don't use create_brach_map in run, use branch_data to transmit required info
- hist_config is not used -> remove the line